src/commonMain/kotlin/io/github/detekt/sarif4k/SarifSerializer.kt

```diff
@@ -15,4 +20,57 @@ object SarifSerializer {
     fun toJson(sarif: SarifSchema210): String = json.encodeToString(sarif)
 
     fun fromJson(json: String): SarifSchema210 = Json.decodeFromString(json)
-}
\ No newline at end of file
+}
+
+// Extracted from https://github.com/Kotlin/kotlinx.serialization/issues/296#issuecomment-1859572541
+private fun Any?.toJsonElement(): JsonElement = when (this) {
```

Comment:
I would put these into a separate file (toJsonElement.kt), because right now they're private implementation details, housed together with THE public API. And the public API is only deeply indirectly calling these.